ajax_with_jquery_exercise/nacker_hews_AJAX/app.js

```diff
@@ -0,0 +1,292 @@
+$(document).ready(function() {
+  let $topiclist = $(".topiclist");
+  let $favorite = $(".favoritetoggle");
+  let $all = $(".alltoggle");
+  let $topStories = $(".top-stories");
+  let $bestStories = $(".best-stories");
+  let $newStories = $(".new-stories");
+  let $login = $(".logtoggle");
+  let $logout = $(".logout");
+  let $loginForm = $(".log-in");
+  let $signUpForm = $(".sign-up");
+  let storage = [];
+
+  let hackerNews = [];
+
+  //Function to fetch top 25 results for fetch news.
+  function fetchTopNews() {
+    $topiclist.empty();
+    $.get("https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty")
+    .then(function(res){
+      hackerNews = hackerNews.concat(res.slice(0,25));
+      hackerNews.forEach(function(value) {
+        $.get("https://hacker-news.firebaseio.com/v0/item/" + value + ".json?print=pretty")
+        .then(function(res){
+          console.log(res);
```

Comment:
no biggie now, but when you're eventually making PRs to get code into a production codebase, make sure you remove console logs beforehand.